### PR TITLE
Fix pending change after flushing `RegisterView`

### PR DESCRIPTION
**Pull request by jvff**
## Motivation

@afck ran into an issue where an assertion caught unexpected pending changes after voting for a block proposal. After investigating the issue, the cause seems to be that `RegisterView::flush` had an edge case where it wouldn't clear it's `update` field.

## Proposal

Clear the `update` field after successfully flushing.

## Test Plan

A unit test was added to reproduce the issue.

## Release Plan

- These changes should be backported to the latest `devnet` branch, then
    - be released in a validator hotfix.
- These changes should be backported to the latest `testnet` branch, then
    - be released in a validator hotfix.

## Links

- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)

### CODE REVIEW

**Review by afck — APPROVED** [review]
Thank you! :tada: :pray:

File: linera-views/src/views/register_view.rs [hunk 1]
```diff
@@ -99,6 +99,7 @@ where
             self.stored_value = value;
         }
         self.delete_storage_first = false;
+        self.update = None;
```
Comment (ma2bd) [inline linera-views/src/views/register_view.rs:102]: Crazy that this bug survived so long
